app/models/katello/content_view.rb

```diff
@@ -453,7 +453,26 @@ def computed_module_ids_by_repoid
     def check_ready_to_publish!
       fail _("User must be logged in.") if ::User.current.nil?
       fail _("Cannot publish default content view") if default?
+      unless composite_action_allowed?(organization.library)
+        fail _("This composite view cannot be published until all of the component content view versions "\
+            "have been published to %{env}.  This restriction is optional and can be modified in the "\
+            "Administrator -> Settings page using the restrict_composite_view flag.") %
+            { :env => organization.library.name }
+
+
+      end
       check_distribution_conflicts!
+      true
+    end
+
+    def composite_action_allowed?(env)
```

Comment:
This should probably get its own unit test or be moved into the private section if it's not intended to be part of the public interface of this model.